src/main/java/auto_scheduler/CoursePrioritizationServiceImpl.java

```diff
@@ -10,16 +13,193 @@
+        }
+        return graph;
+    }
+
+
+    private Dictionary<String, Integer> getCourseMap(
+        Collection<CurriculumCourse> candidateCourses
+    ) {
+        int i = 0;
+        Dictionary<String, Integer> map = new Hashtable<String,Integer>(); 
+        for(CurriculumCourse course: candidateCourses) {
+            String courseNumber = course.getCourseNumber();
+            courseNumber.toUpperCase();
+            map.put(courseNumber, i);
+            i++;
+        }
+        return map;
+    }
+
+
+    // private Dictionary<String, Collection<String>> removePassedClasses(
+    //     Dictionary<String, Collection<String>> courseToPreReqTable,
+    //     Collection<String> passedCourses,
+    //     Integer npreReqs
+    // ) {
+    //     Dictionary<String, >
+    //     if( courseToPreReqTable != null) {
+    //         Collection<String> keys = courseToPreReqTable.getKeys();
+    //         for(String key: keys ) {
+    //             Collection<String> preReqs = courseToPreReqTable.getValue(key);
+                
+    //             Collection<String>  preReqsNotPassed = preReqs.stream().filter(
+    //                 preReq ->
+    //                 passedCourseCodes.stream().noneMatch(
+    //                     passedCourse -> passedCourse.equalsIgnoreCase(preReq)
+    //                 )
+    //             ).collect(Collectors.toList());
+
+    //             courseToPreReqTable.put(key, preReqsNotPassed);
+    //         }
+    //     }
+    // }
+
+    // private Dictionary<String, Collection<String>> getDependencyTable() {
+    //     Dictionary<String, Collection<String>> map = new Hashtable<String, Collection<String>>();
+    //     int[][] graph = new int[n][n];
+    //     int i = 0;
+    //     for(CurriculumCourse course: candidateCourses) { 
+    //         String courseNumber = course.getCourseNumber();
+    //         String preReqString = course.getPreRequisite();
+    //         String[] preReqs = null;
+    //         if(preReqString != null) {
+    //             preReqString = preReqString.trim();
+    //             if(!preReqString.isEmpty()) {
+    //                 preReqs = preReqString.split(":");
+    //             }                    
+    //         }
+    //         if(preReqs != null && preReqs.length != 0) {
+    //             map.put(courseNumber,preReqs);
+    //         }
+            
+    //     }
+    // }
+
+    // private Dictionary<String, String> removePassedClasses(
+    //     Dictionary<String, String> map, 
+    //     Collection<String> passedCourses) {
+    //     if(passedCourses != null) {
+    //         for(String passedCourse: passedCourses) {
+    //             map.remove(passedCourse.toUpperCase());
+    //         }
+    //     }
+    //     return map;
+    // }
+
+
+    // private Dictionary<String, Integer> getUniquePrereqs(
+    //     Collection<CurriculumCourse> candidateCourses,
+    //     Dictionary<String, Integer> candidateCourseMap
+    // ) {
```

Comment:
saving the dead code in case when i am writing the tests i realize there is something wrong.